host/lg/0021-keyboard-input-support-for-multi-windows.patch

```diff
@@ -0,0 +1,182 @@
+From b47b51713187f308807a457296e94f95d20e206d Mon Sep 17 00:00:00 2001
+From: RajaniRanjan <rajani.ranjan@intel.com>
+Date: Wed, 1 Dec 2021 11:35:06 +0530
+Subject: [PATCH] keyboard input support for multi-windows.
+
+keyboard virtual input device support for PGP
+
+Tracked-On: OAM-100023
+Signed-off-by: RajaniRanjan <rajani.ranjan@intel.com>
+---
+ client/include/vInputClient.h |  4 ++--
+ client/src/main.c             | 31 ++++++++++++++++++++++---------
+ client/src/vInputClient.c     | 13 +++++++++++--
+ 3 files changed, 35 insertions(+), 13 deletions(-)
+
+diff --git a/client/include/vInputClient.h b/client/include/vInputClient.h
+index ad81f14d..0bf0b276 100644
+--- a/client/include/vInputClient.h
++++ b/client/include/vInputClient.h
+@@ -1,11 +1,11 @@
+ #define XRES_MAX 600.00
+ #define YRES_MAX 960.00
+ #define LTRB_Y   292.00
+-typedef enum {PRESS, RELEASE, MOVE} action;
++typedef enum {PRESS, RELEASE, MOVE, KEYDOWN, KEYUP} action;
+ //typedef enum {ROTATION_0, ROTATION_1} rotation;
+ 
+ int initvInputClient(const char []);
+ void send_sync_event();
+ void send_event(int32_t type, int32_t code, int32_t value);
+-bool vinput_touch(action a,bool fullscreen, u_int32_t x, u_int32_t y, int rotation, u_int32_t dstRecth,u_int32_t dstRectw,u_int32_t dstRectx);
++bool vinput_touch(action a,bool fullscreen, u_int32_t x, u_int32_t y, int rotation, u_int32_t dstRecth,u_int32_t dstRectw,u_int32_t dstRectx,u_int32_t keys);
```

Comment:
add space after ,

Reply:
Instead of multiple function arguments, you could define and add structure as an argument.

typedef struct vinputEvent {
    action a;
    bool fullscreen;
    u_int32_t x;
    u_int32_t y;
    int rotation;
    u_int32_t dstRecth;
    u_int32_t dstRectw;
    u_int32_t dstRectx;
    u_int32_t keys;
} vinputEvent;

bool vinput_touch(vinputEvent ev);
bool